src/modules/new-request-form/fields/attachments/Attachments.tsx

```diff
@@ -0,0 +1,192 @@
+import {
+  FileUpload,
+  Field as GardenField,
+  Hint,
+  Input,
+  Label,
+  Message,
+} from "@zendeskgarden/react-forms";
+import {
+  Close,
+  Notification,
+  Title,
+  useToast,
+} from "@zendeskgarden/react-notifications";
+import { useCallback } from "react";
+import { useDropzone } from "react-dropzone";
+import type { Field } from "../../data-types";
+import type { AttachedFile } from "./AttachedFile";
+import { FileListItem } from "./FileListItem";
+import { useAttachedFiles } from "./useAttachedFiles";
+
+interface AttachmentProps {
+  field: Field;
+}
+
+async function fetchCsrfToken() {
```

Comment:
Maybe we could turned this into a `useCsrfToken ` reusable hook and memoize it so we only get it once independently of where it's used.

Reply:
I am not sure how the CSRF token works. Does it expire? For example, if the user opens a page and fetches the CSRF token, then it stays idle for 1 hour, and then he starts an upload, is the token still valid?

If the answer is yes, we can use a hook and memoize it. Otherwise, it is better to always fetch it before using it.

Reply:
It is based on the session so in theory in your example it should still be valid as long as the session is alive.

> Otherwise, it is better to always fetch it before using it.

My only concern is if we need several times, e.g. upload and form submission, it may lead to unnecessary calls.
Totally fine to address it later on as well 👍🏼

Reply:
If we can safely fetch it once it is better to memoize it to avoid unnecessary calls, but let's revisit it later when we'll need to use it in another places